plugin/openassetio_manager_bal/bal.py

```diff
@@ -330,6 +330,17 @@ def _copy_and_expand_trait_properties(entity_version_dict: dict, library: dict)
     return expanded_dict
 
 
+class MalformedEntityReference(RuntimeError):
+    """
+    An exception raised for a reference to an entity reference that is
+    formatted incorrectly or otherwise malformed according to BALs
+    expectations.
+    """
+
+    def __init__(self, malformed_reason: str, entity_ref: str):
+        super().__init__(f"{malformed_reason} ({entity_ref})")
+
+
```

Comment:
[consider] `bal.py` is meant to represent some existing system, as such, it's not meant to know about OpenAssetIO concepts, with the bridge happening in the Interface class - might be better to define this there instead?

Reply:
To my mind this MalformedEntityReference now is about an existing system, as it's a placeholder for an exception about an entityReference not living up to BALs expectations.
You think a rename might help to to bring it in line with the other exceptions defined in this file, like `MalformedBALReference`?

Reply:
BAL itself doesn't have any expectations on the string though, i _only_ traffics the `EntityInfo` class - that defines its data, the conversion to/from an Entity Reference is done in the interface layer as part of the "bridging to OpenAssetIO concepts"

Reply:
(This is why `_build_entity_ref` and `_parse_entity_ref` are not in `bal.py` - to help enforce the concept of the `BasicAssetLibraryInterface` class being where you bridge your two systems together. eg `bal.py` is logically some closed source independent tool.

Reply:
[05f5109](https://github.com/OpenAssetIO/OpenAssetIO-Manager-BAL/pull/75/commits/05f5109db36e7a0ec73ee9a665a6701b21066407)